File: src/main/java/com/practica/crean/arquitecture/core/usecases/BuscarPorMatricula.java
package com.practica.crean.arquitecture.core.usecases;

import arrow.core.Either;
import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.shared.excepciones.ResourceNotFoundException;

public interface BuscarPorMatricula {

    Estudiante execute(String matricula);

}

File: src/main/kotlin/com/practica/crean/arquitecture/core/usecases/BuscarPorMatricula.kt
package com.example.demo.core.usecases

import arrow.core.Either
import com.example.demo.core.domain.Estudiante
import com.example.demo.shared.excepciones.ResourceNotFoundException

// <>
fun interface BuscarPorMatricula {

    fun execute(matricula: String): Either<ResourceNotFoundException, Estudiante>

}